backend/service/k8s/cache.go

```diff
@@ -45,6 +45,13 @@ func (s *svc) StartTopologyCaching(ctx context.Context, ttl time.Duration) (<-ch
 		go s.startInformers(ctx, name, cs, ttl)
 	}
 
+	go func() {
+		<-ctx.Done()
+		s.log.Info("Closing the kubernetes topologyObjectChan")
+		close(s.topologyObjectChan)
```

Comment:
this will race with the inner stop channel which could lead to write on a closed channel

Reply:
I see, I moved the stop channel up to `StartTopologyCaching` so we can control the shutdowns / channel close all from there.